vowpalwabbit/parser.cc

```diff
@@ -1048,17 +1028,21 @@ void free_parser(vw& all)
     output->currentname.delete_v();
   }
 
+  std::vector<example*> drain_pool;
```

Comment:
It feels like there should be a way to call `for_each` on the example pool so that you don't need to do all this pushing and popping. Then again, that functionality doesn't really make sense in a resource pool data structure, so maybe this is ok...

Reply:
Yeah I thought the same thing... This will eventually go away too when the destructor is able to delete all of the contents (*cough* unions...)